### PR TITLE
feat(path/unstable): support `URL` in `extname()`

### PR DESCRIPTION
**Pull request by iuioiua**
Towards #5537

### CODE REVIEW

File: path/posix/extname.ts [hunk 1]
```diff
@@ -18,26 +18,37 @@ import { isPosixPathSeparator } from "./_util.ts";
...
  * ```
  *
  * @param path The path to get the extension from.
  * @returns The extension (ex. for `file.ts` returns `.ts`).
  */
-export function extname(path: string): string {
+export function extname(path: URL): string;
+export function extname(path: string | URL): string {
+  if (path instanceof URL) {
+    path = path.pathname;
```
Comment (kt3k) [inline path/posix/extname.ts:49]: Can you use `fromFileUrl` instead? This allows other random urls such as `http:`, `data:`, `jsr:`, `node:`, etc, which is not aligned to how `dirname` works with those url types.

Comment (iuioiua) [reply path/posix/extname.ts:49]: What's the harm of using `.pathname`?

Comment (kt3k) [reply path/posix/extname.ts]: In `data:` scheme, the pathname doesn't represent the path, but it represent arbitrary data. The return value of `extname` can be any random value in that case.

File: path/extname.ts [hunk 2]
```diff
@@ -22,6 +22,29 @@ import { extname as windowsExtname } from "./windows/extname.ts";
...
+ *   assertEquals(extname(new URL("file:///home/user/Documents/image.png")), ".png");
+ * }
+ * ```
+ *
+ * @param path Path with extension.
+ * @returns The file extension. E.g. returns `.ts` for `file.ts`.
+ */
+export function extname(path: URL): string;
+export function extname(path: string | URL): string {
+  // deno-lint-ignore no-explicit-any
```
Comment (timreichen) [inline path/extname.ts:48]: Couldn't `any` be avoided with `toString()` or `String(path)`?

Comment (kt3k) [reply path/extname.ts:48]: We like to do `path instanceof URL` check in each implementation. `String(path)` makes that check not working.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**